service-registry/src/main/java/org/apache/servicecomb/serviceregistry/consumer/MicroserviceVersions.java

```diff
@@ -109,13 +110,18 @@ public String getMicroserviceName() {
     return microserviceName;
   }
 
+  @VisibleForTesting
   public Map<String, MicroserviceVersion> getVersions() {
-    return versions;
+    synchronized (lock) {
+      return versions;
+    }
   }
 
   @SuppressWarnings("unchecked")
   public <T extends MicroserviceVersion> T getVersion(String serviceId) {
-    return (T) versions.get(serviceId);
+    synchronized (lock) {
+      return (T) versions.get(serviceId);
```

Comment:
1.3.x do not clear version when set instances, is this modification necessary and can you reproduce the problem?